### PR TITLE
Ch extension of data types

### PR DESCRIPTION
**Pull request by Annettesunday**
### What does this PR do?

Understanding data types in a deeper dimension

### CODE REVIEW

File: _DATA_TYPES/_Destructuring_assignment/main/main.js [hunk 1]
```diff
@@ -0,0 +1,492 @@
...
+/*
+
+Ignore elements using commas
+
+Unwanted elements of the array can also be thrown away via an extra comma
+*/
+
+// second element is not needed
+
+let [firstName, , title] = ["Juluis", "Caesar", "Consul", "of the Roman Republic"];
```
Comment (RufusMbugua) [inline _DATA_TYPES/_Destructuring_assignment/main/main.js:54]: ES6 is pretty powerful

**Review by RufusMbugua — COMMENTED** [review]
LGTM

Do you have any questions so far?

File: _DATA_TYPES/_Date_Time/main/main.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+new Date()
+// Without arguments - create a Date object for the current date and time
+
+let now = new Date();
+
+alert( now ); // shows current date/ time
+
+new Date(milliseconds)
+// Create a Date object with the time equal to number of milliseconds (1/1000 of a second) passed after Jan 1st of 1970 UTC+0
```
Comment (pndemo) [inline _DATA_TYPES/_Date_Time/main/main.js:30]: Try to limit line length to 120 characters for readability.

File: _DATA_TYPES/_Destructuring_assignment/main/main.js [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+/*
+Properties options.title, options.width and options.height are assigned to the corresponding variables.
+The order does not matter
+
+This works too
+*/
+
+// changed the order of properties in let
+
+let {height, width, title} = { title: "Menu", height: 200, width: 100 }
```
Comment (pndemo) [inline _DATA_TYPES/_Destructuring_assignment/main/main.js:220]: Semi-colons at the end of statements though not compulsory are generally a good practice.

File: _DATA_TYPES/_Iterables/main/main.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+    //  triggers 4 times for each character
+    alert(char); // t, e, s, t
+}
+
+// And it work correctly with surrogate pairs
+
+
+let str = '𝒳😂';
+for (let char of str) {
+    alert( char ); // 𝒳, and then 😂
```
Comment (pndemo) [inline _DATA_TYPES/_Iterables/main/main.js:127]: 👍

**Comment by pndemo** [comment]
Just a few things specified above to consider. Generally, looks good to me.

File: _DATA_TYPES/_Date_Time/main/main.js (hunk 2 above)
Comment (Annettesunday) [reply _DATA_TYPES/_Date_Time/main/main.js:30]: Noted!

**Pull request merged**